firedrake/__init__.py

```diff
@@ -100,3 +100,31 @@
 from firedrake._version import get_versions
 __version__ = get_versions()['version']
 del get_versions
+
+# Try to detect threading and either disable or warn user
+# Threading may come from
+# - OMP_NUM_THREADS: openmp,
+# - OPENBLAS_NUM_THREADS: openblas,
+# - MKL_NUM_THREADS: mkl,
+# - VECLIB_MAXIMUM_THREADS: accelerate,
+# - NUMEXPR_NUM_THREADS: numexpr
+# We only handle the first two cases
+from ctypes import cdll
+from ctypes.util import find_library
+try:
+    _openblas_lib = find_library('openblas')
+    _openblas_dll = cdll.LoadLibrary(_openblas_lib)
+    _openblas_dll.openblas_set_num_threads(1)
```

Comment:
Does this function return an error code we can check?

Reply:
Sadly not:
https://github.com/xianyi/OpenBLAS/blob/develop/driver/others/openblas_set_num_threads.c

Reply:
Of course not, because it just calls OpenMP and https://www.openmp.org/spec-html/5.0/openmpsu110.html#x147-6380003.2.1